File: gitdot-api/src/endpoint/repository/delete_repository_commit_filter.rs
use serde::{Deserialize, Serialize};

use crate::endpoint::Endpoint;

pub struct DeleteRepositoryCommitFilter;

impl Endpoint for DeleteRepositoryCommitFilter {
    const PATH: &'static str = "/repository/{owner}/{repo}/commit-filters/{filter_id}";
    const METHOD: http::Method = http::Method::DELETE;

    type Request = DeleteRepositoryCommitFilterRequest;
    type Response = DeleteRepositoryCommitFilterResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct DeleteRepositoryCommitFilterRequest {}

pub type DeleteRepositoryCommitFilterResponse = ();

File: gitdot-web/packages/gitdot-api/src/endpoint/repository/delete-repository-commit-filter.ts
import { z } from "zod";
import type { Endpoint } from "../endpoint";

export const DeleteRepositoryCommitFilterRequest = z.object({});
export type DeleteRepositoryCommitFilterRequest = z.infer<
  typeof DeleteRepositoryCommitFilterRequest
>;

export const DeleteRepositoryCommitFilterResponse = z.void();
export type DeleteRepositoryCommitFilterResponse = z.infer<
  typeof DeleteRepositoryCommitFilterResponse
>;

export const DeleteRepositoryCommitFilter = {
  path: "/repository/{owner}/{repo}/commit-filters/{filter_id}",
  method: "DELETE",
  request: DeleteRepositoryCommitFilterRequest,
  response: DeleteRepositoryCommitFilterResponse,
} as const satisfies Endpoint;
export type DeleteRepositoryCommitFilter = typeof DeleteRepositoryCommitFilter;
